ts/indicators/atr.src.ts

```diff
@@ -0,0 +1,215 @@
+/* *
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+import H from '../parts/Globals.js';
+
+declare global {
+    namespace Highcharts {
+        class AtrIndicator extends SmaIndicator {
+            public data: Array<AtrIndicatorPoint>
+            public yData: Array<Array<number>>;
+            public getValues(
+                series: Series,
+                params: AtrInidicatorParamsOptions
+            ): (boolean|IndicatorValuesObject);
+        }
+
+        interface AtrIndicatorOptions extends SmaIndicatorOptions {
+            params?: AtrInidicatorParamsOptions;
+        }
+
+        interface AtrInidicatorParamsOptions {
+            period?: number;
+        }
+
+        class AtrIndicatorPoint extends SmaIndicatorPoint {
+            public series: AtrIndicator
+        }
+
+        interface SeriesTypesDictionary {
+            atr: typeof AtrIndicator;
+        }
+    }
+}
+
+
+import U from '../parts/Utilities.js';
+var isArray = U.isArray;
+
+var seriesType = H.seriesType,
+    UNDEFINED: undefined;
+
+/* eslint-disable valid-jsdoc */
+// Utils:
+/**
+ * @private
+ */
+function accumulateAverage(
+    points: Array<Array<(number|Array<number>)>>,
```

Comment:
Maybe you can use the type `Array<[number|Array<number>]>` to narrow it down.

Reply:
Having a hard time to get this to work
```
Argument of type '[number, number[]]' is not assignable to parameter of type '[number | number[]]'.
  Types of property 'length' are incompatible.
    Type '2' is not assignable to type '1'.ts(2345)
``
Any clues?

Reply:
You have to use the tuple everywhere in the module. You are mixing `[number , Array<number>]` (length = 2) with `Array<(number|Array<number>)>` (length = 1).

Reply:
> Maybe you can use the type Array<[number|Array<number>]> to narrow it down.
Yes, I didn't get the tuple of number or Array<number> thing, but Array<[number,Array<number>]> makes sense :)

Reply:
> Maybe you can use the type Array<[number|Array<number>]> to narrow it down.

Yes, I didn't get the whole tuple of number or Array<number> thing, but `[number, Array<number>] makes sense :)